persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOImpl.kt

```diff
@@ -32,39 +57,84 @@ class MessageDAOImpl(private val queries: MessagesQueries) : MessageDAO {
 
     override suspend fun insertMessage(message: MessageEntity) =
         queries.insertMessage(
-            message.id,
-            message.content,
-            message.conversationId,
-            message.date,
-            message.senderUserId,
-            message.senderClientId,
-            message.status
+            id = message.id,
+            text_body = when (message.content) {
+                is TextMessageContent -> message.content.messageBody
+                else -> null
+            },
+            content_type = message.contentType,
```

Comment:
Can't we do a `when` on `message.content`  to map it to the `ContenType`?

Reply:
but it maps exactly the object type on both vars. What would you do on that `when` clause exactly?

Reply:
I suggestion we remove `contentType` from the `MessageEntity` and do something like this instead:

```kotlin
content_type = contentTypeOf(message.content)


fun contentTypeOf(content: MessageEntityContent): ContentType {
    when(content) {
        is TexMessageContent -> ContentType.TEXT
        is AssetMessageContent -> ContentType.ASSET
    }
}
```

Reply:
This mapping is already done on the `MessageMapper` class

Reply:
But you mean leaving the `ContentType` enum class within the `MessageEntity` class?

Reply:
> This mapping is already done on the `MessageMapper` class

Yes it's done in the mapper in logic right now which is IMO is unnecessary because the `contentType` is only relevant when mapping from `SQLDelightMessage` to `MessageEntity`. I therefore propose to make the `ContentType` internal to the persistence module.

Reply:
As it is now we store the same data in two fields: content and contentType, which is bad since we can theoretically construct invalid messages like:

content = MessageContent.Asset
contentType = ContentType.TEXT

Reply:
👍

Reply:
Ok, now I left only the contentType on the SQLDelightMessage level, and persisted it only to map it manually to the correct MessageContent type.